addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -259,14 +246,6 @@ options.registry.carousel = options.Class.extend({
         this.$target.find('[data-target]').attr('data-target', '#' + this.id);
         this._rebindEvents();
     },
-    /**
-     * @override
-     */
-    onFocus: function () {
-        // Needs to be done on focus, not on start, as all other options are
-        // maybe not all initialized in start
```

Comment:
That comment is probably still true, leave the code, you can just replace with retargetEdition instead of the ugly trigger.

Reply:
I understand the problem but I don't think anything changed regarding the crappy stuff that was made in `onFocus` before. The problem was that one option background/colorpicker/... was not initialized in the `start` yet so changing its target was not possible yet. This may not be the case of the `sizing_y` option you are still retargeting but if this is the case, it is by luck (that the carousel option here is initialized after).

Maybe your retarget method can take a parameter to say if it is only for retargeting options (and not changing the active snippet). So you could leave the `onFocus` stuff here, using that parameter and, actually, the `start` call of retarget does not need to enable another snippet either (only retargeting the sizing_y option to the current slide). If you can find another way to solve the original problem, it is also better.